akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -1765,6 +1770,7 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
     case MemberUp(m)                   => receiveMemberUp(m)
     case MemberExited(m)               => receiveMemberExiting(m)
     case MemberRemoved(m, _)           => receiveMemberRemoved(m)
+    case ClusterShuttingDown           => receiveMemberRemoved(cluster.selfMember)
```

Comment:
and just to clarify, ClusterShuttingDown is published right before the MemberRemoved (of all members). We are not interested in the latter here. https://github.com/akka/akka/blob/aa88d288ac8c31c291f4686fd8d9067fe8bae253/akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala#L689